Bio/SeqIO/PirIO.py

```diff
@@ -144,7 +144,8 @@ def __next__(self):
         line = self._line
         if line is None:
             raise StopIteration
-        stream = self.stream
+        # NamedTemporaryFiles don't support __next__ so create an iter as a workaround
+        stream = iter(self.stream)
```

Comment:
Do you actually need this here? We are not calling `next` on `stream`, and `for line in stream` implicitly calls `iter` on `stream`.

Reply:
line 155 is `next(stream).step()` which breaks in the unit tests when run with `NamedTempFiles`.